storage/src/lib.rs

```diff
@@ -366,6 +366,58 @@ pub async fn last_l1_block_seen(
     Ok(res)
 }
 
+/// Get the last block that has been processed on the previous SL.
+pub async fn last_processed_l2_block_on_the_previous_chain(
+    conn: &mut PgConnection,
+    chain_id: u32,
+) -> Result<u64> {
+    let latency = STORAGE_METRICS.call[&"last_processed_l2_block"].start();
+    let res = sqlx::query!(
+        "
+        SELECT
+          max(l2_block_number)
+        FROM
+          l2_blocks
+        WHERE
+          commit_chain_id != $1
```

Comment:
I think this query will not use the index

Reply:
It wouldn't help, != won't use index

Reply:
So, with backward index, it could work